### PR TITLE
fix mkdirp to opendir

### PR DESCRIPTION
**Pull request by takeaways**
I tried to build my application with it, however it showed me "mkdirp is not a fuction" so I fixed to use in my application.

and it works, thanks

### CODE REVIEW

**Comment by vagrantsn** [comment]
Any updates on this? I am facing the same issue when upgrading from Webpack v4 to v5 and can confirm this PR fixes it :rocket: 

cc @goldhand

File: src/index.js [hunk 1]
```diff
@@ -204,7 +204,7 @@ class SWPrecacheWebpackPlugin {
 
     // use the outputFileSystem api to manually write service workers rather than adding to the compilation assets
     return new Promise((resolve, reject) => {
-      outputFileSystem.mkdirp(path.resolve(filepath, '..'), (mkdirErr) => {
+      outputFileSystem.opendir(path.resolve(filepath, '..'), (mkdirErr) => {
```
Comment (vagrantsn) [inline src/index.js:207]: After upgrading node version to support Webpack 5, I noticed this fix started breaking =/

The correct thing to do here is to use `recursive` option from Node > v10.12, as recommended by [mkdirp](https://www.npmjs.com/package/mkdirp#implementation) and as was made on [Webpack](https://github.com/webpack/webpack/pull/10244). This way, the behavior of creating or only reading the folder (when it already exists) is maintained.

```suggestion
      outputFileSystem.mkdir(path.resolve(filepath, '..'), { recursive: true }, (mkdirErr) => {
```

**Review by vagrantsn — COMMENTED** [review]
Me again @nomadGeonilJang! Just pointing out something I just noticed =)

**Comment by takeaways** [comment]
@vagnervst thank you for doing this for me.

**Comment by vagrantsn** [comment]
@nomadGeonilJang I didn't. The referenced commit you saw here was one I made on another repo, intended to be a temporary fix while your PR isn't merged. You can reopen this pull request with my suggested fix as it is still useful =)

**Comment by tomaszczechowski** [comment]
for me this works `outputFileSystem.opendir(path.resolve(filepath, '..')`
I experience the issue after migrating from webpack 4 to 5.

Please merge this PR.

CC @goldhand

**Pull request closed without merging**